test/unit/math/rev/core/var_test.cpp

```diff
@@ -220,6 +220,17 @@ TEST_F(AgradRev, var_vector_views) {
   EXPECT_FLOAT_EQ(A_v.adj()(3) - 3, A_coeff1.adj());
 }
 
+TEST_F(AgradRev, var_matrix_view_assignment) {
+  Eigen::MatrixXd A(4, 4);
+  A << 0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15;
+  stan::math::var_value<Eigen::MatrixXd> A_v(A);
+  A_v.block(0, 0, 3, 3) = A_v.block(1, 1, 3, 3);
+  stan::math::sum(stan::math::from_var_value(A_v)).grad();
+  Eigen::MatrixXd deriv(4, 4);
+  deriv << 0, 0, 0, 1, 0, 1, 1, 2, 0, 1, 1, 2, 1, 2, 2, 2;
+  EXPECT_MATRIX_FLOAT_EQ(A_v.adj(), deriv);
+}
```

Comment:
Adding `stan::math::var_value<Eigen::MatrixXd> A_block = A_v.block(0, 0, 3, 3);` fails with
```
test/unit/math/rev/core/var_test.cpp: In member function ‘virtual void AgradRev_var_matrix_view_assignment_Test::TestBody()’:
test/unit/math/rev/core/var_test.cpp:228:61: error: conversion from ‘var_value<Eigen::Block<Eigen::Map<Eigen::Matrix<double, -1, -1>, 8, Eigen::Stride<0, 0> >, -1, -1, false>>’ to non-scalar type ‘var_value<Eigen::Matrix<double, -1, -1>>’ requested
  228 |   stan::math::var_value<Eigen::MatrixXd> A_block = A_v.block(0, 0, 3, 3);
```

I think we just need a little constructor to do that

```cpp
  template <typename S, require_convertible_t<S&, value_type>* = nullptr,
            require_any_not_plain_type_t<T, S>* = nullptr>
  var_value(const var_value<S>& other) : vi_(new vari_type(other.vi_->val_)) {
    reverse_pass_callback(
        [this_vi = this->vi_, other_vi = other.vi_]() mutable {
          this_vi->adj_ += other_vi->adj_;
        });
  }
```

Reply:
Fixed and added a test.